nexus_id_spec.md

```diff
@@ -0,0 +1,147 @@
+for the source address (for inbound messages).
+
+## Special Authority ID Values
+
+As specified in the last section, Nexus Channel Core requires that the link layer
+is able to broadcast a message, or send it to every other device on the local
+network. This is indicated by a special Authority ID in the Nexus ID that is reserved
+to make this requirement and other common manufacturing use cases easy to support.
+
+The reserved Authority IDs are:
+
+* **0xFFFF**: reserved for development/testing (anyone can use or make IDs in this
+space in a pre-production/testing context)
+* **0x0000-0x00FF**: reserved for core 'GOGLA-related' use cases currently, including a
+'globally unique' PAYG ID scheme
+* **0xFE80**: reserved for dynamically-assigned Nexus IDs (e.g. Nexus IDs that aren't
+globally-unique, e.g. if the link layer has a scheme where it dynamically assigns
+identities to devices that have no factory assigned ID or outside world mapping to those
+device IDs)
+* **0xFFxx (except 0xFFFF)**: reserved to indicate that this message should be broadcast
+to all devices in the local network. As such, it will only be used as a destination
+address. The lower byte is variable 'xx' so that in the future it is possible to add scope
+context to the broadcast. However, it is expected that in most cases, the link layer will
+not support broadcast scopes and ignore the lower byte (the API also includes a 'broadcast'
+flag so the destination address can be completely ignored by the link layer in that case
+as well). This broadcast may elicit zero or one responses from connected devices, implying
+that the broadcast contains content that is understood by one specific device. A link layer
+*may* support broadcasting with responses from all recipients, which can speed up resource
+discovery, but this is not required.
+
+### IPV6 Semantics
+
+The special Authority ID values 0xFE80 and 0xFFxx were chosen to correspond to published
+IPV6 standards. `0xFE80` is the [link-local unicast address](https://tools.ietf.org/html/rfc4291#section-2.4).
+`0xFFxx` includes the "variable-scope" multicast addresses that are currently
+[registered with the IANA](https://www.iana.org/assignments/ipv6-multicast-addresses/ipv6-multicast-addresses.xhtml#variable).
+
+Examples:
+
+* `{0x0000, 0x12345678} // globally-unique GOGLA identifiers; globally-unique Nexus ID`
+* `{0xFE80, 0x00000010} // dynamically-assigned by link layer, valid only in the local system and not globally-unique`
+
+## Internet Addressing and IPV6 Expansion
+
+For devices that are exposed to the Internet, Nexus ID can be expanded to a valid IPV6
+address. The size of the Nexus ID as 48 bits was selected to allow conversion of the Nexus ID into
+a valid IPV6 address using EUI-64 expansion (same as MAC address expansion, described 
+in [RFC 2373](https://tools.ietf.org/html/rfc2373#page-19)).
+
+There is also a granted ARIN CIDR block of IPV6 address space for devices that comply to the
```

Comment:
@chadnorvell @ricehornet 

We need to decide if we will offer this "if you have a Nexus ID, we guarantee we will give you a free IPV6 address thats globally valid" or not.

It doesn't cost us anything (since we've already bought the ARIN block and will continue to pay for it, I assume), but it *is* potentially valuable to someone who might want to connect their device to the internet with a unique IPV6 address?

cc @bsilverthorn

Reply:
Per our conversation today (with @ricehornet), I think we should leave out references to the IPv6 block for now. Until there's a meaningful application for it available, I'm afraid it will serve more to confuse rather than excite.